.kokoro/build-with-run.sh

```diff
@@ -32,6 +32,44 @@ popd
 export GOOGLE_CLOUD_PROJECT=nodejs-docs-samples-tests
 pushd github/nodejs-docs-samples/${PROJECT}
 
+# Verify changes worth testing.
+SIGNIFICANT_CHANGES="$(git --no-pager diff --name-only main..HEAD | grep -Ev '(\.md$|^\.github)' || true)"
```

Comment:
**Thought:**
It would be cool if we could easily extend this list of files being ignored (`(\.md$|^\.github)`).
I want to also include files such as `.gitignore`, `LICENSE`, `CODEOWNERS`.
Just a thought.

Reply:
Extending the list of ignored files makes sense to me. Making it much more easily extensible I'm going to hold off.

My intention in this PR is to find a minimally sufficient stopgap for the weight of Kokoro testing until we can sort out the overall CI direction for the repo.